src/tests/Parser/DisciplinePriest/Features/AtonementAttribution.test.js

```diff
@@ -0,0 +1,30 @@
+import AtonementAttribution from 'Parser/DisciplinePriest/Modules/Features/AtonementAttribution';
+import SPELLS from 'common/SPELLS';
+import processEvents from '../Fixtures/processEvents';
+
+import {
+  TwoSuccesiveDamagingEventsWithAtonementOnSelfBetween,
+  TwoSuccesiveDamagingEvents
+} from '../Fixtures/ReorderingScenarios';
+
+describe('DisciplinePriest.Reordering', () => {
+
+  let atonementAtribution;
+  beforeEach(() => {
+    atonementAtribution = new AtonementAttribution({
+      reorder: () => true,
+      toPlayer: () => true,
+      byPlayer: () => true,
+      toPlayerPet: () => false,
+      byPlayerPet: () => false,
+    });
+  });
+
+  it('Two Succesive Damaging Events with atonement on self between is reordered', () => {
+    let result = atonementAtribution.reorderEvents(TwoSuccesiveDamagingEventsWithAtonementOnSelfBetween);
+    expect(result[2].type).toBe("damage");
```

Comment:
This is pretty hard to understand. If you move the actual array to this method as proposed in the previous comment, would it be an idea to extract the interesting events from the array into temp variables and check those instead?

e.g.

```jsx
expect(result[2]).toBe(firstDamageEvent);
expect(result[3]).toBe(firstSelfHealing);
```

Reply:
Actually best case might be to move all events to temp vars and just check a new array. Example:

```jsx
const firstAtonementSelfHealEvent = { type: 'heal', sourceID: thisPlayer, targetID: otherPlayer1, amount: amount, timestamp: timestamp, ability: { guid: SPELLS.ATONEMENT_HEAL_NON_CRIT.id } };
const firstDamagingEvent = { type: 'damage', sourceID: thisPlayer, targetID: enemy, amount: amount, timestamp: timestamp, ability: { guid: SPELLS.PURGE_THE_WICKED_BUFF.id } };
const secondAtonementSelfHealEvent = { type: 'heal', sourceID: thisPlayer, targetID: thisPlayer, amount: amount, timestamp: timestamp, ability: { guid: SPELLS.ATONEMENT_HEAL_NON_CRIT.id } };
const secondDamagingEvent = { type: 'damage', sourceID: thisPlayer, targetID: enemy, amount: amount, timestamp: timestamp, ability: { guid: SPELLS.SHADOWBIND_TRAIT.id } };
const firstAtonementAllyHealEvent = { type: 'heal', sourceID: thisPlayer, targetID: otherPlayer1, amount: amount, timestamp: timestamp, ability: { guid: SPELLS.ATONEMENT_HEAL_NON_CRIT.id } };
const secondAtonementAllyHealEvent = { type: 'heal', sourceID: thisPlayer, targetID: otherPlayer2, amount: amount, timestamp: timestamp, ability: { guid: SPELLS.ATONEMENT_HEAL_NON_CRIT.id } };

const originalEvents = [
  firstAtonementSelfHealEvent,
  firstDamagingEvent,
  secondAtonementSelfHealEvent,
  secondDamagingEvent,
  firstAtonementAllyHealEvent,
  secondAtonementAllyHealEvent,
];

expect(atonementAtribution.reorderEvents(TwoSuccesiveDamagingEventsWithAtonementOnSelfBetween)).toEqual([
  firstDamagingEvent,
  secondDamagingEvent,
  firstAtonementSelfHealEvent,
  firstAtonementAllyHealEvent,
  secondAtonementSelfHealEvent,
  secondAtonementAllyHealEvent,
]);

```

I reckon this makes the expected input -> transformation -> output the easiest to see.

Reply:
I agree 100%, will change it shortly